python/app/dialog.py

```diff
@@ -1104,9 +1165,11 @@ def setup_entity_model_view(self, entity_data):
         self._app.log_debug("Creating %r..." % ModelClass)
 
         # create model
-        entity_data["model"] = ModelClass(
-            entity_data["entity_type"], entity_data["view"], self._task_manager
-        )
+        if not entity_data["has_filter_menu"]:
```

Comment:
Instead of checking for the filter menu, we should just check if the model key-value already exists. If it does not exist, then set it, else leave it.